core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java

```diff
@@ -234,7 +236,6 @@ private void workProcessorOperatorStateMonitor(WorkProcessor.ProcessState<Page>
             long deltaReadTimeNanos = deltaAndSet(context.readTimeNanos, sourceOperator.getReadTime().roundTo(NANOSECONDS));
 
             context.dynamicFilterSplitsProcessed.set(sourceOperator.getDynamicFilterSplitsProcessed());
-            context.metrics.set(sourceOperator.getMetrics());
```

Comment:
Shouldn't this move be in second commit?

Reply:
squashed commits